### DISCUSSION TITLE
Is there a way to replay messages from the JetStream staring from some sequence number?

### DISCUSSION THREAD

**Discussion by vol-ale** [nats-io/nats.net.v1#827] [Q&A] [answered]
I'm quite new to NATS and haven't found this information in the examples.
Our consumer uses windowing functions, and in case of consumer restart, it should be able to replay the last N messages. Is such a scenario supported? I do not plan to acknowledge messages - just keep a sequence number.

**Comment by scottf** [comment, marked as answer]
When you create your consumer you can use 
```
ConsumerConfiguration.Builder()
    .WithDeliverPolicy(DeliverPolicy.ByStartSequence)
    .WithStartSequence(42)
    .WithAckPolicy(AckPolicy.None)
    ...
    .Build();
```

The start sequence value would be the stream sequence + 1 of the last message you read. You can find the stream sequence in the Msg.MetaData.StreamSequence.
